### ISSUE TITLE
[DICT] Add Python support for copy

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#1881]
### Describe the feature
Syntactic and semantic support should be added for `copy`. This should be tested by implementing Python printing

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
def foo():
    a = {1,1.0, 2:2.0}
    b = a.copy()
    return b
```

**Comment by EmilyBourne** [comment]
Hi @Joao-antonio-gg, thanks for your interest in Pyccel. This issue is very similar to #1740 which was solved in #1765. You can have a look at that PR to get an idea of what needs to be done. The main steps are:
- [ ] Create a class to represent the `copy` method in the API in https://github.com/pyccel/pyccel/blob/devel/pyccel/ast/builtin_methods/dict_methods.py (you can find documentation about this step here: https://github.com/pyccel/pyccel/blob/devel/developer_docs/ast_nodes.md)
- [ ] Declare the method in the `ClassDef` object in https://github.com/pyccel/pyccel/blob/devel/pyccel/ast/class_defs.py
- [ ] Add tests in https://github.com/pyccel/pyccel/blob/devel/tests/epyccel/test_epyccel_dicts.py
- [ ] Update the docs in https://github.com/pyccel/pyccel/blob/devel/docs/builtin-functions.md#dictionary-methods

Once you're done you can open a PR and the bot should guide you through the [review process](https://github.com/pyccel/pyccel/blob/devel/developer_docs/review_process.md)

Does this sound ok to you? If so then I will assign the issue to you.

P.S your English sounds fine ;)

**Comment by Joao-antonio-gg** [comment]
Yes, the sound is good, i will start now. Thanks
